source/shadow.c

```diff
@@ -148,8 +148,8 @@ static ShadowStatus_t validateMatchTopicParameters( const char * pTopic,
  * @param[in]  thingNameLength Length of Thing Name string pointed to by pThingName.
  * @param[in]  pShadowName Shadow Name string.
  * @param[in]  shadowNameLength Length of Shadow Name string pointed to by pShadowName.
- * @param[in] pTopicBuffer Pointer to buffer for returning the topic string.
- * @param[in] pOutLength Pointer to caller-supplied memory for returning the length of the topic string.
+ * @param[in]  pTopicBuffer Pointer to the topic buffer.
+ * @param[in]  pOutLength Pointer to the length of the topic created.
```

Comment:
Can this be changed to an integer value instead of a pointer?

Reply:
No. The validation is done for pointers in this function.